### PR TITLE
[SYCL][Doc] Clean up contribution guide

### PR DESCRIPTION
**Pull request by bader**
- Limit lines to 80 characters
- Make links to the repo files relative
- Remove outdated information

Signed-off-by: Alexey Bader <alexey.bader@intel.com>

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -63,32 +62,38 @@ Then you just add a line to every git commit message:
...
-Merge of pull request is done only by project maintainers. There are three options:
- - [Rebase and merge] The preferable choice for PRs containing a single commit.
- - [Squash and merge] Used when there are multiple commits in the PR. 
-   Squashing is done to shorten history and make sure that the project is buildable on any commit.
- - [Create a merge commit] Used for pull down PRs to avoid duplication of
-   LLVM commits.
+Project maintainers merge pull requests using one of the following options:
+- [Rebase and merge] The preferable choice for PRs containing a single commit
+- [Squash and merge] Used when there are multiple commits in the PR
+  - Squashing is done make sure that the project is buildable on any commit
```
Comment (AGindinson) [inline CONTRIBUTING.md]:
```suggestion
  - Squashing is done make to sure that the project is buildable on any commit
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -63,32 +62,38 @@ Then you just add a line to every git commit message:
...
-     - **sycl-ubu-x64-pr** - runs all LIT tests on machine with OpenCL runtimes for CPU and GPU.
- - Approval is lost once the PR branch is updated. New approval and checks rerun are required.
- - Once approval is received and all checks pass, the pull request is ready to be merged.
+- Create a pull request for your changes following [Creating a pull request
+instructions](https://help.github.com/articles/creating-a-pull-request/)
+- When the pull request is created signed-off check is done
+  - **check_pr** - verifies that the signed-off mark is added to each commit
+  message
+- Once the pull request is approved by an Intel representative, build and
+functional testing are done
```
Comment (AGindinson) [inline CONTRIBUTING.md]: AFAIU, this behavior has changed (the build/testing tasks are launched regardless of the approval). Or is the procedure different for our external contributors?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -63,32 +62,38 @@ Then you just add a line to every git commit message:
...
- - Create a pull request for your changes following [Creating a pull request instructions](https://help.github.com/articles/creating-a-pull-request/).
- - When the pull request is created signed-off check is done.
-     - **check_pr** - verifies that the signed-off mark is added to each commit message.
- - Once the pull request is approved by an Intel representative, build and functional testing are started.
-     - **sycl-ubu-x64-pr** - runs all LIT tests on machine with OpenCL runtimes for CPU and GPU.
- - Approval is lost once the PR branch is updated. New approval and checks rerun are required.
- - Once approval is received and all checks pass, the pull request is ready to be merged.
+- Create a pull request for your changes following [Creating a pull request
+instructions](https://help.github.com/articles/creating-a-pull-request/)
+- When the pull request is created signed-off check is done
```
[the remark this answers is not in the document]
Comment (bader) [reply CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md (hunk 1 above)
Comment (bader) [reply CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md (hunk 2 above)
Comment (bader) [reply CONTRIBUTING.md]: Yes.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -63,32 +62,38 @@ Then you just add a line to every git commit message:
...
+  - **check_pr** - verifies that the signed-off mark is added to each commit
+  message
+- Once the pull request is approved by an Intel representative, build and
+functional testing are done
+- Approval is reset once the PR branch is updated
+  - New approval and checks rerun are required
+- Once PR is approved and all checks pass, the pull request is ready for merge
```
Comment (pvchupin) [inline CONTRIBUTING.md]: I would prefer the following text here:

- CI will run signed-off check as soon as PR is created, see **check_pr** CI check for results
- CI will run build and functional testing check as soon as PR is approved by Intel representative
  - New approval is needed if PR was updated (e.g. during code review)
- Once PR is approved and all checks pass, the pull request is ready for merge

Comment (bader) [reply CONTRIBUTING.md]: Done.

**Comment by bader** [comment]
@pvchupin, ping.

**Pull request merged**